neo/Implementations/Wallets/peewee/UserWallet.py

```diff
@@ -102,18 +126,23 @@ def OnCreateAccount(self, account):
 
         db_account, created = Account.get_or_create(
             PrivateKeyEncrypted=encrypted_pk, PublicKeyHash=account.PublicKeyHash.ToBytes())
-        db_account.PrivateKeyEncrypted = encrypted_pk
+        if not created:
+            db_account.PrivateKeyEncrypted = encrypted_pk
         db_account.save()
         self.__dbaccount = db_account
 
     def AddContract(self, contract):
+        """
+        Add a contract to the database.
+
+        Args:
+            contract(neo.SmartContract.Contract): a Contract instance.
+        """
         super(UserWallet, self).AddContract(contract)
 
         db_contract = None
         try:
             db_contract = Contract.get(ScriptHash=contract.ScriptHash.ToBytes())
-            db_contract.delete_instance()
```

Comment:
@localhuman please feedback: has there been a reason to delete the old contract here? Not sure this change is compatible.

Reply:
for what it is worth, the new code matches closer to https://github.com/neo-project/neo/blob/master/neo/Implementations/Wallets/EntityFramework/UserWallet.cs#L37-L41
(also note https://github.com/neo-project/neo/blob/master/neo/Implementations/Wallets/EntityFramework/UserWallet.cs#L59 where they actually save the context regardless if it took the if or else branch, which I'm doing now as well).  Oh and the tests didn't fail! ;)

Reply:
Thanks for the explanation. I'd still be interested if @localhuman knows of a reason this code was here in the first place, and if this change works.

Reply:
I like deleting the existing instance so that any FKs get cleaned out, but if the new version works that's ok too!

Reply:
@ixje please restore the line here.